zeus/metric.py

```diff
@@ -288,28 +319,36 @@ def begin_window(self, name: str) -> None:
                 self.update_period,
                 self.prometheus_url,
                 self.job,
+                sync_execution,
             ),
         )
         self.proc.start()
         if not self.proc.is_alive():
             raise RuntimeError(f"Failed to start monitoring process for {name}.")
 
-    def end_window(self, name: str) -> None:
+        self.window_state[name] = MonitoringProcessState(
+            queue=self.queue, proc=self.proc
+        )
```

Comment:
This is kind of mind-blowing. Why are you putting these in `self.queue` and `self.proc`??